services/endpointsrv/service.go

```diff
@@ -54,6 +79,14 @@ type Store interface {
 	EndpointCheckStepUpdate(ctx context.Context, arg db.EndpointCheckStepUpdateParams) error
 	EndpointCheckSteps(ctx context.Context, checkID string) ([]db.UnweaveEndpointCheckStep, error)
 	EndpointCheck(ctx context.Context, checkID string) (db.UnweaveEndpointCheck, error)
+
+	EndpointVersion(ctx context.Context, versionID string) (db.UnweaveEndpointVersion, error)
+	EndpointVersionCreate(ctx context.Context, arg db.EndpointVersionCreateParams) error
+	EndpointVersionList(ctx context.Context, endpointID string) ([]db.UnweaveEndpointVersion, error)
+	EndpointVersionDemote(ctx context.Context, endpointID string) error
+	EndpointVersionPromote(ctx context.Context, id string) error
+
+	Tx(txFunc func(db.Querier) error) error
```

Comment:
We need to decide whether the store returns the db types or the domain types. The original plan was for these to be domain types so we didn't leak db abstractions. I'm fine with whatever is more practical though. We probably shouldn't mix them though.